internal/cmd/generator/tasks/tasks.go

```diff
@@ -255,26 +287,22 @@ var testPageTmpl = template.Must(template.New("testPage").Parse(`// Code generat
 package task
 
 import (
-	"errors"
 	"testing"
 
-	"github.com/Xuanwo/navvy"
 	"github.com/stretchr/testify/assert"
 
+	"github.com/qingstor/noah/pkg/schedule"
 	"github.com/qingstor/noah/pkg/types"
-	"github.com/qingstor/noah/pkg/fault"
 )
 
-var _ navvy.Pool
-var _ types.Pool
 `))
 
 var testTmpl = template.Must(template.New("test").Funcs(funcs).Parse(`
 func Test{{ .Name }}Task_TriggerFault(t *testing.T) {
 	task := &{{ .Name }}Task{}
-	task.SetFault(fault.New())
-	err := errors.New("test error")
-	task.TriggerFault(err)
-	assert.True(t, task.GetFault().HasError())
+	task.SetScheduler(schedule.NewScheduler())
```

Comment:
Remove `Scheduler` interface and `RealScheduler`, returns a struct names `Scheduler` maybe better?

After that, we can use `schedule.New()`